### PR TITLE
V4 DDO typings

### LINKED ISSUE #585: DDO typings for V4 in ocean.py
Create V4 asset class in ocean.py that respects the specs added here.
Specs for DDO: 
https://github.com/oceanprotocol/docs/blob/feature/ddo_v4/content/concepts/did-ddo.md#compute-resources

Aqua: https://github.com/oceanprotocol/aquarius/blob/shacl2/aquarius/ddo_checker/shacl_schemas/v4/remote_4.0.0.ttl
Ocean.js: https://github.com/oceanprotocol/ocean.js/pull/1111

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes #585  .

Changes proposed in this PR:

- created a draft Asset class for V4;
- created the Service class for V4;
- updated `ServiceTypes`.

### CODE REVIEW

**Comment by mariacarmina** [comment]
I will close this PR due to unwanted commits from previous PR.

File: ocean_lib/assets/v4/asset.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            "chainId": self.chain_id,
+        }
+
+        data["metadata"] = self.metadata
+
+        if self.services:
+            data["service"] = [service.as_dictionary() for service in self.services]
+        if self.credentials:
+            data["credentials"] = self.credentials
+        # TODO: NFT data
```
Comment (calina-c) [inline ocean_lib/assets/v4/asset.py:83]: should this TODO be addressed in this PR?

File: ocean_lib/common/agreements/service_types.py [hunk 2]
```diff
@@ -20,3 +20,16 @@ class ServiceTypesIndices:
...
     DEFAULT_ACCESS_INDEX = 3
     DEFAULT_COMPUTING_INDEX = 4
+
+
+class ServiceTypesV4:
+    """Types of Service allowed in ocean protocol DDO services for V4."""
+
+    ASSET_ACCESS = "access"
+    CLOUD_COMPUTE = "compute"
+    AUTHORIZATION = "wss"
```
Comment (calina-c) [inline ocean_lib/common/agreements/service_types.py:30]: is there such a service (authorization)?

File: ocean_lib/services/v4/service.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+from enforce_typing import enforce_types
+
+from ocean_lib.common.agreements.service_types import ServiceTypesV4, ServiceTypesNames
+
+
+@enforce_types
+class NFTService:
+    """Service class to create validate service in a V4 DDO."""
+
+    SERVICE_TYPE = "type"
```
Comment (calina-c) [inline ocean_lib/services/v4/service.py:20]: I would ditch these constants. The usage seems limited and they seem to impair readability.

**Review by calina-c — CHANGES_REQUESTED** [review]

**Pull request closed without merging**